src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -197,10 +198,45 @@ public bool IsSetAsOutput
         /// </summary>
         public string Name
         {
-            get { return nodeLogic.Name; }
+            get 
+            {
+                if (OriginalName != nodeLogic.Name)
+                    IsRenamed = true;
+
+                return nodeLogic.Name; 
+            }
             set { nodeLogic.Name = value; }
         }
 
+        /// <summary>
+        /// The original name of the node.
+        /// This is the name of the node as it is when first added to the canvas.
+        /// </summary>
+        public string OriginalName
+        {
+            get { return OriginalNameFromCreationName(); }
+        }
+
+        /// <summary>
+        /// Construct the original name from the creation name
+        /// by removing the unwanted part of the creation name.
+        /// ex Autodesk.Designscript.Geometry.Point.ByCoordinate@double,double,double
+        /// would remove everyting after the "@" and the take the last two words from Autodesk.Designscript.Geometry.Point.ByCoordinate
+        /// </summary>
+        /// <returns></returns>
+        private string OriginalNameFromCreationName()
+        {
+            string[] longName = nodeLogic.CreationName.Split('@')[0].Split('.');
+            string[] typeAndMethodName = longName.Skip(Math.Max(0, longName.Count() - 2)).ToArray();
+            return String.Join(".", typeAndMethodName);
+        }
+
+        public bool IsRenamed
```

Comment:
Do we really need to serialize this info or we can get from a runtime comparison?

Reply:
changing the schema to include additional info is a headache - if it's possible to not serialize this and determine it based on a runtime comparison this would be useful.

Reply:
+1 in favour of runtime comparison and maybe doing this with a method instead of field/prop. @SHKnudsen Could we look into that please ?